src/zenml/zen_stores/schemas/pipeline_run_schemas.py

```diff
@@ -225,58 +225,36 @@ def to_model(
         }
         config = PipelineConfiguration.parse_raw(self.pipeline_configuration)
 
-        if _block_recursion:
-            return PipelineRunResponseModel(
-                id=self.id,
-                name=self.name,
-                workspace=self.workspace.to_model(),
-                user=self.user.to_model(True) if self.user else None,
-                schedule_id=self.schedule_id,
-                orchestrator_run_id=self.orchestrator_run_id,
-                start_time=self.start_time,
-                end_time=self.end_time,
-                status=self.status,
-                pipeline_configuration=config,
-                num_steps=self.num_steps,
-                git_sha=self.git_sha,
-                client_version=self.client_version,
-                server_version=self.server_version,
-                client_environment=client_environment,
-                orchestrator_environment=orchestrator_environment,
-                created=self.created,
-                updated=self.updated,
-                metadata=metadata,
-            )
-        else:
-            return PipelineRunResponseModel(
-                id=self.id,
-                name=self.name,
-                stack=self.stack.to_model() if self.stack else None,
-                workspace=self.workspace.to_model(),
-                user=self.user.to_model(True) if self.user else None,
-                orchestrator_run_id=self.orchestrator_run_id,
-                start_time=self.start_time,
-                end_time=self.end_time,
-                status=self.status,
-                pipeline=(
-                    self.pipeline.to_model(False) if self.pipeline else None
-                ),
-                build=self.build.to_model() if self.build else None,
-                deployment=self.deployment.to_model()
-                if self.deployment
-                else None,
-                schedule_id=self.schedule_id,
-                pipeline_configuration=config,
-                num_steps=self.num_steps,
-                git_sha=self.git_sha,
-                client_version=self.client_version,
-                server_version=self.server_version,
-                client_environment=client_environment,
-                orchestrator_environment=orchestrator_environment,
-                created=self.created,
-                updated=self.updated,
-                metadata=metadata,
-            )
+        pipeline = self.pipeline.to_model() if self.pipeline else None
+        build = self.build.to_model() if self.build else None
+        deployment = self.deployment.to_model() if self.deployment else None
+        steps = steps or {}
+
+        return PipelineRunResponseModel(
```

Comment:
So this is always being hydrated now?

Reply:
Yes, but actually less things get hydrated now because of this change:
- Previously, `_block_recursion` was only used for pipeline run models inside of pipeline models (since hydrating them would hydrate the pipeline again, leading to infinite loop circular hydration)
- Now, the pipeline model doesn't have any runs hydrated into it anymore since `pipeline.runs` is now a property. Therefore, we no longer need the entire `_block_recursion` logic that I removed above.